File: 7.Sorting/BubbleSort/Bubblesort.py
import time

def bubbleSort(arr):
    n = len(arr)
    for i in range(n):
        for j in range(0, n - i - 1):
            if arr[j] > arr[j + 1]:
                arr[j], arr[j + 1] = arr[j + 1], arr[j]
    return arr
arr = [64, 34, 25, 12, 22, 11, 90]
print("Original array:", arr)
start_time = time.time()
sorted_arr = bubbleSort(arr)
end_time = time.time()
print("Sorted array:", sorted_arr)
print(f"Time taken to execute: {end_time - start_time:.6f} seconds")

File: 7.Sorting/BubbleSort/BubbleSort.c
// Bubble Sort in C

#include <stdio.h>
#include <stdlib.h>
#include <time.h>

void swap(int *a, int *b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

void BubbleSort(int arr[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {
        for (int j = 0; j < n - i - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                swap(&arr[j], &arr[j + 1]);
            }
        }
    }
}

void display(int arr[], int size)
{
    for (int i = 0; i < size; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
}

int main()
{
    int A[100000], n;
    clock_t start, end;

    printf("Enter n: ");
    scanf("%d", &n);
    for (int i = 0; i < n; i++)
    {
        A[i] = rand() % 10000; 
    }
    printf("Original array:\n");
    display(A, n);

    start = clock();
    BubbleSort(A, n);
    end = clock();

    printf("Sorted array:\n");
    display(A, n);

    double time_taken = (double)(end - start) / CLOCKS_PER_SEC;
    printf("Time taken to sort: %lf seconds\n", time_taken);

    return 0;
}
